abc/quantities.clp

```diff
@@ -29,7 +28,9 @@ process(S,T) :- process(_,S,T).
 :- shelved(S,A,N), &sum { shelved(S,A,T); process(A,O,S,T) : limit(A,O,S,M) } != shelved(S,A,T-1), time(T).
 
 
-:- process(O,S,T), target(O,P), position(P,C), not position(S,C,T-1).
+:- process(O,S,T), target(O,P), position(P,(X,Y)), &sum{ positionX(S,T-1)}!=X.
+:- process(O,S,T), target(O,P), position(P,(X,Y)), &sum{ positionY(S,T-1)}!=Y.
+
```

Comment:
a design decision, or..?

Reply:
The old rule doesn't work, because you need the position of the shelve as a numeric value using positionX and positionY and not position/3 (which is only defined for the initial time step). Like this you can actually restrict that it is on the same position as the picking station for any other time step.